resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -793,6 +793,25 @@ private[spark] class Client(
       // distributed file.
       amKeytabFileName.foreach { kt => props.setProperty(KEYTAB.key, kt) }
 
+      // Upload user provided ivysettings.xml file to the distributed cache
+      val ivySettings = sparkConf.getOption("spark.jars.ivySettings")
+      if (isClusterMode && ivySettings.isDefined) {
+        val ivySettingsFile = new File(ivySettings.get)
+        require(ivySettingsFile.exists(), s"Ivy settings file $ivySettingsFile not found")
```

Comment:
This would be a little more Scala-idiomatic for unpacking the `Option`:
```
      ivySettings match {
        case Some(ivySettingsPath) if isClusterMode =>
          // ... logic here
        case _ => // do nothing
      }
```

Reply:
also check to see if Utils.isLocalUri

Reply:
Done